applications/asset_tracker/src/main.c

```diff
@@ -766,6 +788,10 @@ void cloud_event_handler(const struct cloud_backend *const backend,
 		printk("CLOUD_EVT_PAIR_DONE\n");
 		on_pairing_done();
 		break;
+	case CLOUD_EVT_FOTA_DONE:
+		printk("CLOUD_EVT_FOTA_DONE\n");
+		sys_reboot(SYS_REBOOT_COLD);
```

Comment:
I'm wondering if we should make a function for rebooting "gracefully", as we're now rebooting from several places in the code. My concern is that logs will only be displayed if `CONFIG_LOG_IMMEDIATE` is enabled, otherwise we'll reboot before the logging thread has time to process the pending log messages. `LOG_PANIC()` can be called before reboot to let the logging backend handle pending items.

Reply:
While outside the scope of this PR, I think this is a great idea